modules/csg/csg_shape.cpp

```diff
@@ -167,78 +169,244 @@ void CSGShape3D::_make_dirty(bool p_parent_removing) {
 	dirty = true;
 }
 
-CSGBrush *CSGShape3D::_get_brush() {
-	if (dirty) {
-		if (brush) {
-			memdelete(brush);
+enum {
```

Comment:
I don't like unnamed enums. We should rename this to `ManifoldProperty`.

```suggestion
enum ManifoldProperty {
```

I also tried changing this to `enum class` but since these enum values are used as an integers in most places that would be worse in this case.